packages/@aws-cdk/custom-resource-handlers/lib/aws-stepfunctions-tasks/cross-region-aws-sdk-handler/index.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+/* eslint-disable no-console */
+
+type Event = {
+  /**
+   * AWS Region name to call the service
+   */
+  region: string;
+
+  /**
+   * AWS service name (must be in AWS SDK for JS v3 style)
+   */
+  service: string;
+
+  /**
+   * AWS API endpoint
+   */
+  endpoint: string;
+
+  /**
+   * API Action name (must be in camelCase or PascalCase)
+   */
+  action: string;
+
+  /**
+   * request parameters to call the API (must be in AWS SDK for JS v3 style)
+   */
+  parameters?: any;
+};
+
+// reference: code from CDK AwsCustomResource
+function findV3ClientConstructor(pkg: object) {
+  const [_clientName, ServiceClient] = Object.entries(pkg).find(([name]) => {
+    // Services expose a base __Client class that we don't want ever
+    return name.endsWith('Client') && name !== '__Client';
+  }) as [
+    string,
+    {
+      new (config: any): {
+        send: (command: any) => Promise<any>;
+        config: any;
+      };
+    },
+  ];
+  return ServiceClient;
+}
+
+function findCommandClass(pkg: object, action: string) {
+  const camelToPascal = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
```

Comment:
Small nit, but you can move the `camelToPascal` function outside the `findCommandClass` scope, to prevent it being re-declared on every call